### PR TITLE
Feature 17 emoji picker refactor

### PR DESCRIPTION
**Pull request by vvainio**
Attempt at DRYing things up, still lots of room for improvement.

### CODE REVIEW

File: addon/services/emoji.js [hunk 1]
```diff
@@ -107,71 +98,31 @@ export default Service.extend({
...
-  travel__tone_5:   filterBy('emoji__tone_5', 'category', 'travel'),
-  objects__tone_5:  filterBy('emoji__tone_5', 'category', 'objects'),
-  symbols__tone_5:  filterBy('emoji__tone_5', 'category', 'symbols'),
-  flags__tone_5:    filterBy('emoji__tone_5', 'category', 'flags'),
-  regional__tone_5: filterBy('emoji__tone_5', 'category', 'regional'),
-  modifier__tone_5: filterBy('emoji__tone_5', 'category', 'modifier'),
+  init() {
+    this._defineEmojiCategoryProperties();
+    this._defineEmojiToneProperties();
+  },
```
Comment (lolmaus) [inline addon/services/emoji.js]: Should this contain `this._super()`?

File: addon/components/emoji-picker.js [hunk 2]
```diff
@@ -5,12 +5,26 @@ import {debounce, next, throttle} from 'ember-runloop';
...
 import {A} from 'ember-array/utils';
 import {default as EObject} from 'ember-object';
-const  O = EObject.create.bind(EObject);
+import {EMOJI_CATEGORIES_ARRAY, EMOJI_TONES_ARRAY} from "../utils/constants";
+
+const O = EObject.create.bind(EObject);
+const EMOJI_PICKER_SCROLLABLE_ELEMENT = '.eeo-emojiPicker-scrollable';
+const DEPENDENT_KEYS_FOR_EMOJI_CATEGORIES = EMOJI_CATEGORIES_ARRAY.join(',');
+const DEPENDENT_KEYS_FOR_EMOJI_SERVICE = (() => {
+  let keys = [];
```
Comment (lolmaus) [inline addon/components/emoji-picker.js]: Can this be `const`?

File: addon/services/emoji.js (hunk 1 above)
Comment (vvainio) [reply addon/services/emoji.js]: Nope, services don't require it:
> "By default, does nothing unless it is overridden during class definition."
http://emberjs.com/api/classes/Ember.Service.html#method_init

File: addon/components/emoji-picker.js (hunk 2 above)
Comment (vvainio) [reply addon/components/emoji-picker.js]: Yes! 👍

File: addon/components/emoji-picker.js [hunk 3]
```diff
@@ -5,12 +5,26 @@ import {debounce, next, throttle} from 'ember-runloop';
...
+
+  EMOJI_CATEGORIES_ARRAY.forEach(category => {
+    EMOJI_TONES_ARRAY.forEach(tone => {
+      const key = `emojiService.${category}__tone_${tone}`;
+      dependentKeys.push(key);
+    });
+  });
+
+  return dependentKeys.join(',');
+})();
```
Comment (lolmaus) [inline addon/components/emoji-picker.js]: Should we move `DEPENDENT_KEYS_FOR_EMOJI_CATEGORIES` and `DEPENDENT_KEYS_FOR_EMOJI_SERVICE` into `constants.js`?

Comment (vvainio) [reply addon/components/emoji-picker.js]: Not sure, I actually had them there initially but moved them here as they're only used in this certain component.

Comment (lolmaus) [reply addon/components/emoji-picker.js]: I'm in favor of keeping them in `constants.js`.

Comment (vvainio) [reply addon/components/emoji-picker.js]: Yep, totally fine for me. I'll move them 👍

**Review by lolmaus — APPROVED** [review]

**Pull request merged**